include/stronk/unit.hpp

```diff
@@ -78,12 +84,24 @@ struct unit
          * @tparam NewUnitValueT The new unit value type to convert to with same dimensions and underlying type
          * @return a new unit with same dimensions and underlying type, but with the specified scale
          */
-        template<unit_value_like NewUnitValueT>
-            requires(std::same_as<typename NewUnitValueT::unit_t::dimensions_t, dimensions_t>
-                     && std::same_as<UnderlyingT, typename NewUnitValueT::underlying_type>)
-        constexpr auto to() const -> NewUnitValueT
+        template<template<typename UT> typename NewUnitValueT>
```

Comment:
I don't think you have named UT in other places where a template type is also templated.

E.g. in stronk skills, it is `template<typename> typename... Skills`